streams/src/main/java/org/apache/kafka/streams/kstream/Repartitioned.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream;
+
+import org.apache.kafka.clients.producer.internals.DefaultPartitioner;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.kstream.internals.WindowedSerializer;
+import org.apache.kafka.streams.kstream.internals.WindowedStreamPartitioner;
+import org.apache.kafka.streams.processor.StreamPartitioner;
+
+/**
+ * This class is used to provide the optional parameters for internal repartitioned topics when using:
+ * <ul>
+ *     <li>{@link KStream#repartition(Repartitioned)}</li>
+ *     <li>{@link KStream#repartition(KeyValueMapper, Repartitioned)}</li>
+ * </ul>
+ *
+ * @param <K> key type
+ * @param <V> value type
+ */
+public class Repartitioned<K, V> implements NamedOperation<Repartitioned<K, V>> {
+
+    protected final String name;
```

Comment:
Thanks for avoiding mutable state in this class!

Reply:
@mjsax I've created JIRA ticket for it: https://issues.apache.org/jira/browse/KAFKA-9342